feature/experimental/p4rt/ate_tests/performance_test/performance_test.go

```diff
@@ -0,0 +1,932 @@
+		WithDefaultGateway(dutPort2.IPv4)
+	i2.IPv6().
+		WithAddress(atePort2.IPv6CIDR()).
+		WithDefaultGateway(dutPort2.IPv6)
+
+	return top
+}
+
+// configureDeviceIDs configures p4rt device-id on the DUT.
+func configureDeviceID(t *testing.T, dut *ondatra.DUTDevice) {
+	nodes := p4rtutils.P4RTNodesByPort(t, dut)
+	p4rtNode, ok := nodes["port1"]
+	if !ok {
+		t.Fatal("Couldn't find P4RT Node for port: port1")
+	}
+	t.Logf("Configuring P4RT Node: %s", p4rtNode)
+	c := oc.Component{}
+	c.Name = ygot.String(p4rtNode)
+	c.IntegratedCircuit = &oc.Component_IntegratedCircuit{}
+	c.IntegratedCircuit.NodeId = ygot.Uint64(deviceID)
+	gnmi.Replace(t, dut, gnmi.OC().Component(p4rtNode).Config(), &c)
+}
+
+// setupP4RTClient sends client arbitration message for the leader client,
+// then sends setforwordingpipelineconfig config.
+func setupP4RTClient(args *testArgs) error {
+	// Setup p4rt-client stream parameters.
+	streamParameter := p4rt_client.P4RTStreamParameters{
+		Name:        streamName,
+		DeviceId:    deviceID,
+		ElectionIdH: uint64(0),
+		ElectionIdL: electionID,
+	}
+
+	// Send ClientArbitration message from the leader client.
+	clients := []*p4rt_client.P4RTClient{args.leader}
+	for index, client := range clients {
+		if client != nil {
+			client.StreamChannelCreate(&streamParameter)
+			if err := client.StreamChannelSendMsg(&streamName, &p4_v1.StreamMessageRequest{
+				Update: &p4_v1.StreamMessageRequest_Arbitration{
+					Arbitration: &p4_v1.MasterArbitrationUpdate{
+						DeviceId: streamParameter.DeviceId,
+						ElectionId: &p4_v1.Uint128{
+							High: streamParameter.ElectionIdH,
+							Low:  streamParameter.ElectionIdL - uint64(index),
+						},
+					},
+				},
+			}); err != nil {
+				return fmt.Errorf("errors seen when sending ClientArbitration message: %v", err)
+			}
+			if _, _, arbErr := client.StreamChannelGetArbitrationResp(&streamName, 1); arbErr != nil {
+				if err := p4rtutils.StreamTermErr(client.StreamTermErr); err != nil {
+					return err
+				}
+				return fmt.Errorf("errors seen in ClientArbitration response: %v", arbErr)
+			}
+		}
+	}
+
+	// Load p4info file.
+	p4Info, err := utils.P4InfoLoad(p4InfoFile)
+	if err != nil {
+		return errors.New("Errors seen when loading p4info file.")
+	}
+
+	// Send SetForwardingPipelineConfig for p4rt leader client.
+	if err := args.leader.SetForwardingPipelineConfig(&p4_v1.SetForwardingPipelineConfigRequest{
+		DeviceId:   deviceID,
+		ElectionId: &p4_v1.Uint128{High: uint64(0), Low: electionID},
+		Action:     p4_v1.SetForwardingPipelineConfigRequest_VERIFY_AND_COMMIT,
+		Config: &p4_v1.ForwardingPipelineConfig{
+			P4Info: &p4Info,
+			Cookie: &p4_v1.ForwardingPipelineConfig_Cookie{
+				Cookie: 159,
+			},
+		},
+	}); err != nil {
+		return errors.New("Errors seen when sending SetForwardingPipelineConfig.")
+	}
+	return nil
+}
+
+// getGDPParameter returns GDP related parameters for testPacketOut testcase.
+func getGDPParameter() PacketIO {
```

Comment:
Go style says to not start function names with "get".
https://google.github.io/styleguide/go/decisions#getters
These can just be "gdpParameter", "tracerouteParameter", "lldpParameter".